### PR TITLE
IOSXR bgp address family

### PR DESCRIPTION
**Pull request by ashwini-mhatre**
##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/iosxr_bgp_address_family.py [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+        address_family:
+          description: Enable address family and enter its config mode
+          type: list
+          elements: dict
+          suboptions:
+            afi:
+              description: address family.
+              type: str
+              choices: ['ipv4', 'ipv6', 'l2vpn', 'link-state', 'vpnv4', 'vpnv6']
+            af_modifier:
```
Comment (NilashishC) [inline plugins/modules/iosxr_bgp_address_family.py:42]:
```suggestion
            safi:
```

File: tests/integration/targets/iosxr_bgp_address_family/tests/common/merged.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+---
+- debug:
+    msg: Start iosxr_bgp_address-family merged integration tests connection={{ ansible_connection}}
+
+- include_tasks: _remove_config.yaml
+
+- block:
+    - name: Merge the provided configuration with the exisiting running configuration
```
Comment (NilashishC) [inline tests/integration/targets/iosxr_bgp_address_family/tests/common/merged.yaml:8]: Can we have a couple of integration tests with VRF?

**Review by NilashishC — CHANGES_REQUESTED** [review]

File: plugins/modules/iosxr_bgp_address_family.py [hunk 3]
```diff
@@ -0,0 +1,614 @@
...
+              description: VRF name.
+              type: str
+            additional_paths: &additional_paths
+              description: BGP additional-paths commands
+              type: str
+              choices: [ 'send', 'receive' ]
+            advertise_best_external: &advertise
+              description: Advertise best-external path.
+              type: bool
+            aggregate_address: &aggrigate_address
```
Comment (NilashishC) [inline plugins/modules/iosxr_bgp_address_family.py:56]: This should be a list, shouldn't it?

File: plugins/modules/iosxr_bgp_address_family.py [hunk 4]
```diff
@@ -0,0 +1,614 @@
...
+                    max_path_value:
+                      type: int
+                      description: <2-64>  Number of paths (limit includes backup path).
+                    order_igp_metric:
+                      description: Order candidate multipaths for selection as per configured number(cisco-support).
+                      type: bool
+                    selective_order_igp_metric:
+                      description: Allow multipaths only from marked neighbors
+                      type: bool
+            network: &network
```
Comment (NilashishC) [inline plugins/modules/iosxr_bgp_address_family.py:267]: This should also be a list.

File: plugins/modules/iosxr_bgp_address_family.py [hunk 5]
```diff
@@ -0,0 +1,614 @@
...
+                        ebgp:
+                          type: int
+                          description: Update limit for eBGP sub-groups<1-512.
+                    address_family:
+                      type: int
+                      description: Update limit for sub-groups.
+                wait_install:
+                  type: bool
+                  description: Wait for route install.
+            redistribute: &redistribute
```
Comment (NilashishC) [inline plugins/modules/iosxr_bgp_address_family.py:343]: This should also be a list.

Comment (NilashishC) [reply plugins/modules/iosxr_bgp_address_family.py:343]: Please refer to how redistribute is handled from either bgp or ospf modules.

**Review by NilashishC — CHANGES_REQUESTED** [review]
Please make sure that all keys that should be lists are identified as such. It becomes a challenge to change the type of a key once the module is shipped.

File: plugins/modules/iosxr_bgp_address_family.py (hunk 3 above)
Comment (ashwini-mhatre) [reply plugins/modules/iosxr_bgp_address_family.py]: ack,made changes

File: plugins/modules/iosxr_bgp_address_family.py (hunk 5 above)
Comment (ashwini-mhatre) [reply plugins/modules/iosxr_bgp_address_family.py]: ack,made changes

File: plugins/modules/iosxr_bgp_address_family.py (hunk 1 above)
Comment (ashwini-mhatre) [reply plugins/modules/iosxr_bgp_address_family.py]: ack,made changes

**Review by NilashishC — APPROVED** [review]

**Pull request merged**